src/safeposix/net.rs

```diff
@@ -97,6 +97,19 @@ impl NetMetadata {
         }
     }
 
+    pub fn _release_localport(&mut self, addr: interface::GenIpaddr, port: u16, protocol: i32, domain: i32) -> Result<i32, i32> {
+        if protocol ==  IPPROTO_UDP && self.used_port_set.contains(&mux_port(addr.clone(), port, domain, UDPPORT)) {
+            self.used_port_set.remove(&mux_port(addr.clone(), port, domain, UDPPORT));
+        }
+        else if protocol ==  IPPROTO_TCP && self.used_port_set.contains(&mux_port(addr.clone(), port, domain, TCPPORT)) {
+            self.used_port_set.remove(&mux_port(addr.clone(), port, domain, TCPPORT));
+        }
+        else {
+            return Err(syscall_error(Errno::EINVAL, "release", "provided port is not being used"));
+        }
+        return Ok(0);
+    }
+
```

Comment:
2 things: 
1. rather than returning a Result<i32, i32> where the Ok is always 0, return a Result<(), i32>.
2. Rather than check that the port set contains a value, call remove on the value and check whether the remove result is not None.

Reply:
Made it so that it checks whether it is true or not. Please let me know if that is what you wanted.